util/tracing/detect/recorder.go

```diff
@@ -70,23 +96,56 @@ func (r *TraceRecorder) Record(traceID trace.TraceID) func() []tracetest.SpanStu
 	}
 }
 
-func (r *TraceRecorder) gc() {
-	r.mu.Lock()
-	defer r.mu.Unlock()
+func (r *TraceRecorder) gcLoop(ctx context.Context) {
+	defer close(r.done)
+
+	ticker := time.NewTicker(time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case now := <-ticker.C:
```

Comment:
There is a difference in here from old code that under load this ticker will not start to slow down.

Reply:
I'm not sure what you mean. I believe ticker automatically drops ticks or slows down for slow receivers: https://pkg.go.dev/time#Ticker

Looking at the old code, it does appear that the old one would have 1 minute as the initial timer and then 50 seconds for every subsequent interval. I'm not sure if that was intentional or not.